### PR TITLE
cli-plugins/hooks: fix max-message off-by-one

### LINKED ISSUE #7005: cli-plugins/hooks: max-message enforcement is off by one
### Description

`cli-plugins/hooks.ParseTemplate()` currently enforces `maxMessages = 10` by counting newline characters and rejecting only when `strings.Count(out, "\n") > maxMessages`, then returning `strings.SplitN(out, "\n", maxMessages)`.

That makes the limit off by one:

- a template that renders to **10 newline characters** actually contains **11 messages**
- that output is currently accepted instead of rejected
- `SplitN(..., 10)` then merges the extra line into the last returned element, so the final "message" can still contain an embedded newline

This means the "maximum 10 messages" guard can be bypassed by one extra rendered line.

### Reproduce

1. Use the current `master` branch.
2. Call `hooks.ParseTemplate()` with output that contains 11 lines, for example:

```go
strings.Repeat("line\n", 10) + "line"
```

3. Observe that no error is returned.
4. Observe that the returned slice has length 10, with the last element containing an embedded newline.

The relevant logic is currently in:

- `cli-plugins/hooks/template.go:43`
- `cli-plugins/hooks/template.go:46`

### Expected behavior

A template that renders to more than 10 messages should be rejected, including the 11-line / 10-newline case.

In other words, the limit should be enforced on the number of returned messages, not just on the number of newline separators.

### docker version

N/A for runtime environment. This is a source-level issue in the current `master` branch checkout.

Observed in local checkout at `9f16882de4`.

### docker info

N/A for runtime environment. This issue is in template parsing logic and does not depend on daemon configuration.

### Additional Info

This looks like it was introduced by `b7ab63387a` (`cli-plugins/hooks: limit maximum number of lines / messages`).

I’m happy to send a fix with a regression test if this behavior matches the intended bug report.

### PR DESCRIPTION
**Pull request by fallintoplace**
Fixes #7005

## What changed

This updates `cli-plugins/hooks.ParseTemplate()` to enforce the 10-message limit on the actual number of rendered messages instead of on the number of newline separators alone.

It also adds a regression test for the 11-line case.

## Why it changed

The current implementation rejects only when `strings.Count(out, "\n") > maxMessages` and then returns `strings.SplitN(out, "\n", maxMessages)`.

That makes the guard off by one: output with 10 newline characters contains 11 messages, but is still accepted. The extra rendered line is then folded into the last returned element.

## Impact

Templates that render to more than 10 messages are now rejected consistently, including the 11-line / 10-newline case.

Valid templates keep the existing split behavior.

## Root cause

The limit check counted separators instead of messages.

## Validation

- `GOPATH=$(mktemp -d) GO111MODULE=off go test github.com/docker/cli/cli-plugins/hooks` via a temporary GOPATH layout rooted at this checkout

### CODE REVIEW

**Review by CZNetworks — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thank you for contributing! It looks like your commit message is missing a DCO sign-off,
causing the DCO check to fail.

We require all commit messages to have a `Signed-off-by` line with your name
and e-mail (see ["Sign your work"](https://github.com/moby/moby/blob/docker-v29.4.1/CONTRIBUTING.md#sign-your-work)
in the CONTRIBUTING.md in this repository), which looks something like:

```
Signed-off-by: YourFirstName YourLastName <yourname@example.org>
```

There is no need to open a new pull request, but to fix this (and make CI pass),
you need to _amend_ the commit(s) in this pull request, and "force push" the amended
commit.

Unfortunately, it's not possible to do so through GitHub's web UI, so this needs
to be done through the git commandline.

You can find some instructions in the output of the DCO check (which can be found
in the "checks" tab on this pull request), as well as in the [Moby contributing guide](https://github.com/moby/moby/blob/v20.10.11/docs/contributing/set-up-git.md).

Steps to do so "roughly" come down to:

1. Set your name and e-mail in git's configuration:

    ```bash
    git config --global user.name "YourFirstName YourLastName"
    git config --global user.email "yourname@example.org"
    ```

    (Make sure to use your _real_ name (**not your GitHub username/handle**) and e-mail)


2. Clone your fork locally
3. Check out the branch associated with this pull request
4. Sign-off and amend the existing commit(s)

    ```bash
    git commit --amend --no-edit --signoff
    ```

    If your pull request contains multiple commits, either squash the commits (if
    needed) or sign-off each individual commit.

5. _Force push_ your branch to GitHub (using the `--force` or [`--force-with-lease`](https://stackoverflow.com/questions/52823692/git-push-force-with-lease-vs-force) flags) to update the pull request.


Let me know if you need help or more detailed instructions!

**Comment by fallintoplace** [comment]
@thaJeztah Thank you. I will check later today.

**Review by vvoland — APPROVED** [review]

**Review by docker-agent — COMMENTED** [review]
### Assessment: 🟢 APPROVE

File: cli-plugins/hooks/template.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func ParseTemplate(hookTemplate string, cmd *cobra.Command) ([]string, error) {
 		}
 		out = b.String()
 	}
-	if n := strings.Count(out, "\n"); n > maxMessages {
+	if n := strings.Count(out, "\n") + 1; n > maxMessages {
```
Comment (docker-agent) [inline cli-plugins/hooks/template.go:43]: **[medium] Trailing-newline output incorrectly rejected: `+1` overcounts messages when output ends with `\n`**

The fix changes the guard from `strings.Count(out, "\n") > maxMessages` to `strings.Count(out, "\n") + 1 > maxMessages`. This is correct for output that does **not** end with a trailing newline, but it overcounts when output does end with `\n`.

**Trigger path:** A template that renders to exactly 10 messages each followed by a newline — e.g., `"msg1\nmsg2\n...\nmsg10\n"` — has 10 newline characters. The new guard computes `n = 10 + 1 = 11 > 10 = maxMessages` and rejects the output with the error `"hook template contains too many messages (11): maximum is 10"`. Yet the template only produced 10 real messages; the 11th "segment" from `strings.Split` is just an empty string after the final `\n`.

**Impact:** Valid 10-message templates whose rendered output ends with a trailing newline (common when Go `text/template` templates place a literal newline after the last field) are incorrectly refused. The regression test added in this PR uses `strings.Repeat("line\n", 10)+"line"` (no trailing newline), so it does not catch this edge case.

**Suggested fix:** Trim a trailing newline before counting, so an empty final segment is not treated as an extra message:

```suggestion
	if n := strings.Count(strings.TrimRight(out, "\n"), "\n") + 1; n > maxMessages {
```

| Confidence | Score |
| :--: | :--: |
| 🟢 strong | 100/100 |

**Review by docker-agent — COMMENTED** [review]
### Assessment: 🟡 NEEDS ATTENTION
